python/cuml/manifold/simpl_set.pyx

```diff
@@ -138,6 +139,14 @@ def fuzzy_simplicial_set(X,
     umap_params.deterministic = <bool> deterministic
     umap_params.set_op_mix_ratio = <float> set_op_mix_ratio
     umap_params.local_connectivity = <float> local_connectivity
+    try:
+        umap_params.metric = metric_parsing[metric.lower()]
+    except KeyError:
+        raise ValueError(f"Invalid value for metric: {metric}")
+    if isinstance(metric_kwds, dict):
+        umap_params.p = <float> metric_kwds.get("p", 2.0)
+    else:
+        umap_params.p = <float> 2.0
```

Comment:
This could lead to a bug where `metric_kwds` is passed with a different type (e.g. legitimately some mapping or illegitimately a completely different type) and then `p` is inexplicably always set 2.0. This would be safer:
```suggestion
    if metric_kwds is None:
        umap_params.p = <float> 2.0
    else:
        umap_params.p = <float> metric_kwds.get("p", 2.0)
```
Alternatively a bit more concise, but not tested and would support falsey arguments:
```python
    umap_params.p = metric_kwds.get("p", 2.0) if metric_kwds else <float> 2.0
```